simulator-core/src/test/java/com/evbox/everon/ocpp/mock/expect/RequestExpectationManager.java

```diff
@@ -0,0 +1,27 @@
+package com.evbox.everon.ocpp.mock.expect;
+
+import com.evbox.everon.ocpp.mock.match.RequestMatcher;
+import com.evbox.everon.ocpp.simulator.message.Call;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Function;
+
+/**
+ * Manages all request expectations.
+ */
+public class RequestExpectationManager extends ExpectationManager<RequestMatcher, Call> {
+
+    /**
+     * Find expected ocpp response associated with the incoming request.
+     * Return {@code Optional.empty()} if nothing was found.
+     *
+     * @param incomingRequest incoming request.
+     * @return optional of expected response.
+     */
+    public Optional<Function<Call, String>> findExpectedResponse(Call incomingRequest) {
+
+        List<RequestMatcher> matchers = getSuccessfulMatchers(incomingRequest);
```

Comment:
Better to return optional directly from getSuccessfulMatchers(). Why do you need a list here?

Reply:
I don't but we might need it in the future, the abstract class should be able to retrieve successful matchers. 

 We could add getFirstSuccessfulMatch or something like that only returns one, if you don't like a list here?